pkg/cmd/roachtest/tests/backup.go

```diff
@@ -209,7 +215,7 @@ func registerBackup(r registry.Registry) {
 
 				// Upload the perf artifacts to any one of the nodes so that the test
 				// runner copies it into an appropriate directory path.
-				if err := c.PutE(ctx, t.L(), t.PerfArtifactsDir(), t.PerfArtifactsDir(), c.Node(1)); err != nil {
+				if err := c.PutE(ctx, t.L(), localPerfDir, t.PerfArtifactsDir(), c.Node(1)); err != nil {
```

Comment:
FYI `PutString` exists, do you even need the local temp file? Seems like you could write to an in-mem buffer instead.
If you're looking to write a file, it might be betterto pick a temp file inside of `t.ArtifactsDir()` to make sure it gets collected as well if things go wrong (i.e. pass `t.ArtifactsDir()` is first param to `TempDir`).

Reply:
Oh, nice! Thanks for the tip. Updated to use PutString.